crates/tags/src/storage/memory.rs

```diff
@@ -26,29 +25,29 @@ impl Default for Memory {
 
 #[async_trait]
 impl Storage for Memory {
-    async fn tags(&self) -> Result<Vec<Tag>> {
+    async fn tags(&self) -> Result<Vec<TagName>> {
         let lock = self.0.read().await;
         Ok(lock.keys().cloned().collect())
     }
 
-    async fn del(&self, tag: Tag) -> Result<()> {
+    async fn del(&self, tag: TagName) -> Result<()> {
         let mut lock = self.0.write().await;
         lock.remove(&tag)
             .ok_or_else(|| Error::from_str(StatusCode::NotFound, ""))?;
         Ok(())
     }
 
-    async fn get(&self, tag: Tag) -> Result<Hash> {
+    async fn get(&self, tag: TagName) -> Result<TagValue> {
         let lock = self.0.read().await;
         let x = lock
             .get(&tag)
             .ok_or_else(|| Error::from_str(StatusCode::NotFound, ""))?;
         Ok(x.clone())
     }
 
-    async fn put(&self, tag: Tag, hash: Hash) -> Result<()> {
+    async fn put(&self, tag: TagName, data: TagValue) -> Result<()> {
```

Comment:
If we store `TagValue`, then we store the value twice - once as an opaque byte vector and second time as a Rust `Kind` structure. We also store the tag name twice, since `TagValue` contains it for some reason.

I think this is the signature we need:
```suggestion
    async fn put(&self, tag: String, data: Vec<u8>) -> Result<()> {
```